### PR TITLE
Improve API sign tests and fix a few minor issues

### PR DESCRIPTION
**Pull request by lziest**
- Test api/sign on subject overriden and hostname overriden
- Fix duplicate code
- Add a test on signer/local of rejecting label in the sign request.

### CODE REVIEW

**Comment by lziest** [comment]
hmm, test failure.

File: README.md [hunk 1]
```diff
@@ -49,13 +49,15 @@ The version command takes no arguments.
 #### Signing
 
 ```
-cfssl sign [-ca cert] [-ca-key key] hostname csr [subject]
+cfssl sign [-ca cert] [-ca-key key] [-hostname hostname] csr [subject]
```
Comment (grittygrease) [inline README.md]: If sign also supports comma-separated hostnames, add it here. Otherwise we should include support by re-using the code from gencert.

Comment (lziest) [inline README.md]: Good point. Fixed. Sign supports comma-separated hostnames already.

**Comment by grittygrease** [comment]
LGTM

**Comment by kisom** [comment]
LGTM.

**Pull request merged**